src/powershell/Public/ConvertTo-FinOpsSchema.ps1

```diff
@@ -0,0 +1,148 @@
+    }
+
+    $csvData = Import-Csv -Path $ActualCost
+    
+    $rowCount = $csvData.Count
+    $processedCount = 0
+    $start = [DateTime]::Now
+    
+    $transformedData = $csvData | ForEach-Object {
+        $object = $_
+    
+        # Create a new object with the mapped column names
+        # This will ensure that the output CSV has the correct column names
+        $newObject = @{
+            BillingAccountId     = $object.BillingAccountId
+            BillingAccountName   = $object.BillingAccountName
+            ftk_AccountType      = (Get-AccountType -Object $object)
+            BillingProfileId     = $object.BillingProfileId
+            SubscriptionId       = $object.SubscriptionId
+            #... other mappings ...
+        }
+
+    # TODO: 
+        # Unexpected Columns: When an unexpected column is encountered, the name of the column is added to $unexpectedColumns to notify the user after processing.
+        # $RemoveCustomColumns: When $RemoveCustomColumns is $false (or not set), the unexpected columns are allowed to pass through to the output data. 
+        # If $RemoveCustomColumns is $true, unexpected columns are omitted from the output. Write-Warning: A single warning with all unexpected column names is issued after processing, avoiding repetitive warnings during processing.
+            
+        $processedCount++
+        
+        # Time Estimation Logic.
+        # If we have processed less than 10 rows, we will use a constant value of .01 seconds per row.
+        # This is to avoid a divide by zero error. After 10 rows, we will use the average time per row.
+        $percent = [Math]::Round(($processedCount / $rowCount) * 100,1)
+        # $secPerRow is the average processing time per row.
+        $secPerRow = if ($processedCount -lt 10) { .01 } else { ([DateTime]::Now - $start).TotalSeconds/$processedCount }
+        # $remaining is the estimated remaining time for the processing of the rest of the data based on that average.
+        $remaining = $secPerRow * ($rowCount - $processedCount)
+
+        # Number Formatting.
+        # We want to format the numbers to be more readable. 
+        # We will use the current culture to determine the appropriate formatting.
+        $formattedProcessedCount = $processedCount.ToString('N0', [System.Globalization.CultureInfo]::CurrentCulture)
+        $formattedRowCount = $rowCount.ToString('N0', [System.Globalization.CultureInfo]::CurrentCulture)
+
+        Write-Progress -Activity "Converting to FinOps Schema" `
+            -Status "Row $formattedProcessedCount of $formattedRowCount, $percent% complete" `
+            -PercentComplete $percent `
+            -SecondsRemaining $remaining
+
+        [PSCustomObject]$newObject
+    }
+
+    if (-not $ExportAllColumns) {
+        $transformedData = $transformedData | Select-Object BillingAccountId,BillingAccountName,SubscriptionId #... other needed properties ...
```

Comment:
I think we can do this even without the columns all there...
```suggestion
        $transformedData = $transformedData | Select-Object AmortizedCost, AvailabilityZone, BilledCost, BillingAccountId, BillingAccountName, BillingCurrency, BillingPeriodEnd, BillingPeriodStart, ChargePeriodEnd, ChargePeriodStart, ChargeType, ProviderName, PublisherName, Region, ResourceId, ResourceName, ServiceCategory, ServiceName, SubAccountId, SubAccountName
```

Reply:
ok